### PR TITLE
fix: rework FastifyErrors, ensure documentation completeness

### PR DESCRIPTION
**Pull request by Uzlopak**
I saw first some issues on the Errors documentation on fastify.dev, like so:
image

So I started fixing them manually. But then I realized, that I am a lazy guy. So I wrote unit tests, to ensure that the documentation is regarding Errors complete. 

Then I realized, that some of those Errors should be TypeErrors, so I also fixed that. 

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by climba03003 — CHANGES_REQUESTED** [review]

File: lib/errors.js [hunk 1]
```diff
@@ -319,12 +332,14 @@ const codes = {
   FST_ERR_INVALID_URL: createError(
     'FST_ERR_INVALID_URL',
     "URL must be a string. Received '%s'",
-    400
+    500,
```
[the remark this answers is not in the document]
Comment (Uzlopak) [reply lib/errors.js:335]: The only place it is called is https://github.com/fastify/fastify/blob/c79b49e2aa2c7e210cb258a78a710376f6e17e38/lib/route.js#L134

So it is actually not an Error a client can get but an error, which happens at startup time. So it is an internal server error and not a Bad Request.

**Review by mcollina — APPROVED** [review]
lgtm

File: lib/errors.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (climba03003) [reply lib/errors.js:335]: I don't mind if it is changed in `next` but changing it as this stage may break others.

[the remark this answers is not in the document]
Comment (Uzlopak) [reply lib/errors.js:335]: I can change that back in this branch, and when next is sync with main i could provide a PR only changing this line

**Review by metcoder95 — APPROVED** [review]
LGTM, I just share the same concern as @climba03003 for the status code change. And, your suggestion SGTM as well 👍

File: test/internals/errors.test.js [hunk 2]
```diff
@@ -591,8 +593,8 @@ test('FST_ERR_INVALID_URL', t => {
   t.equal(error.name, 'FastifyError')
   t.equal(error.code, 'FST_ERR_INVALID_URL')
   t.equal(error.message, "URL must be a string. Received '%s'")
-  t.equal(error.statusCode, 400)
-  t.ok(error instanceof Error)
+  t.equal(error.statusCode, 500)
```
Comment (Uzlopak) [inline test/internals/errors.test.js:596]:
```suggestion
  t.equal(error.statusCode, 400)
```

**Comment by Uzlopak** [comment]
FST_ERR_INVALID_URL is now again status code 400

**Pull request merged**